common/src/autogluon/common/utils/utils.py

```diff
@@ -47,3 +50,62 @@ def setup_outputdir(path, warn_if_exist=True, create_dir=True, path_suffix=None)
     if path[-1] != os.path.sep:
         path = path + os.path.sep
     return path
+
+
+def get_python_version(include_micro=True) -> str:
+    if include_micro:
+        return f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
+    else:
+        return f"{sys.version_info.major}.{sys.version_info.minor}"
+
+
+def get_package_versions() -> Dict[str, str]:
+    """Gets a dictionary of package name -> package version for every package installed in the environment"""
+    import pkg_resources
+    package_dict = pkg_resources.working_set.by_key
+    package_version_dict = {key: val.version for key, val in package_dict.items()}
+    return package_version_dict
+
+
+def get_autogluon_metadata() -> Dict[str, Any]:
+    from ..version import __version__
+    metadata = dict(
+        system=platform.system(),
+        version=f'{__version__}',
+        py_version=get_python_version(include_micro=False),
+        py_version_micro=get_python_version(include_micro=True),
+        packages=get_package_versions(),
+    )
+    return metadata
+
+
+def compare_autogluon_metadata(*, original: dict, current: dict, check_packages=True) -> list:
+    logs = []
+    og = original
+    cu = current
+    if og['version'] != cu['version']:
+        logs.append((30, f"WARNING: AutoGluon version mismatch (original={og['version']}, current={cu['version']})"))
+    if og['py_version'] != cu['py_version']:
+        logs.append((30, f"WARNING: AutoGluon Python version mismatch (original={og['py_version']}, current={cu['py_version']})"))
+    elif og['py_version_micro'] != cu['py_version_micro']:
+        logs.append((30, f"INFO: AutoGluon Python micro version mismatch (original={og['py_version_micro']}, current={cu['py_version_micro']})"))
+    if og['system'] != cu['system']:
+        logs.append((30, f"WARNING: System mismatch (original={og['system']}, current={cu['system']})"))
+    if check_packages:
+        og_pac = og['packages']
+        cu_pac = cu['packages']
+        for k in og_pac.keys():
+            if k not in cu_pac:
+                logs.append((30, f"WARNING: Missing package '{k}=={og_pac[k]}'"))
+            elif og_pac[k] != cu_pac[k]:
+                logs.append((30, f"WARNING: Package version diff '{k}'\t(original={og_pac[k]}, current={cu_pac[k]})"))
+        for k in cu_pac.keys():
+            if k not in og_pac:
+                logs.append((30, f"INFO: New package '{k}=={cu_pac[k]}'"))
```

Comment:
This will produce lots of warnings between models trained on GPU and deployed on CPU. I don't think it's a good idea.

Reply:
This is not enabled by default, it only shows these warnings when check_packages=True (False by default)

Reply:
This is mainly for debugging, if it fails user can load with check_packages=True to see the differences.